overlord/snapshotstate/backend/backend.go

```diff
@@ -476,9 +476,49 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 
 var ErrCannotCancel = errors.New("cannot cancel: import already finished")
 
+// multiError is a helper to handle mutliple errors.
+type multiError struct {
+	header string
+	errs   []error
+}
+
+// newMultiError returns a new multiError struct initialized with
+// the given format string that explains what operation potentially
+// want wrong. multiError can be nested and will render correctly
+// in these cases.
+func newMultiError(header string, errs []error) error {
+	return &multiError{header: header, errs: errs}
+}
+
+// Error formats the error string.
+func (me *multiError) Error() string {
+	return me.nestedError(0)
+}
+
+// helper to ensure formating of nested multiErrors works.
+func (me *multiError) nestedError(level int) string {
+	indent := strings.Repeat(" ", level)
+	buf := bytes.NewBufferString(fmt.Sprintf("%s:\n", me.header))
+	if level > 5 {
+		return "cannot nest multi errors deeper than 5 levels"
```

Comment:
this sounds like the user could help avoid it, where is more of unexpected corner case/internal error. I would go with something like:

"more error nesting than expected (max 10 levels)??"